server/service/software.go

```diff
@@ -27,14 +29,27 @@ func listSoftwareEndpoint(ctx context.Context, request interface{}, svc fleet.Se
 	if err != nil {
 		return listSoftwareResponse{Err: err}, nil
 	}
-	return listSoftwareResponse{Software: resp}, nil
+
+	var latest time.Time
+	for _, sw := range resp {
+		if !sw.CountsUpdatedAt.IsZero() && sw.CountsUpdatedAt.After(latest) {
+			latest = sw.CountsUpdatedAt
+		}
+	}
+	return listSoftwareResponse{CountsUpdatedAt: latest, Software: resp}, nil
 }
 
 func (svc Service) ListSoftware(ctx context.Context, opt fleet.SoftwareListOptions) ([]fleet.Software, error) {
 	if err := svc.authz.Authorize(ctx, &fleet.Software{}, fleet.ActionRead); err != nil {
 		return nil, err
 	}
 
+	// default sort order to hosts_count descending
+	if opt.OrderKey == "" {
```

Comment:
```
| order_key               | string  | query | What to order results by. Can be ordered by the following fields: `name`, `hosts_count`. Defaults to the hosts count, descending.                                                                                                                                                                                                           |
| order_direction         | string  | query | **Requires `order_key`**. The direction of the order given the order key. Options include `asc` and `desc`. Default if not provided is `asc`.   
```
^ re `order_direction` docs: If `order_key` is not provided, then  the provided`order_direction` is ignored and set to `desc`.

Reply:
Ah, I missed the `**Requires `order_key`**` :)

Reply:
Should we err if the caller provides `opt.OrderDirection` but not `opt.OrderKey`?

Reply:
Currently there's no way to detect if one was provided, because the zero value of `OrderDirection` is `OrderAscending`. If we want to enforce this check, we should do it consistently on all endpoints with a `ListOptions` (but we could introduce errors in existing calls - the UI or scripts in the wild - that otherwise had this situation just ignored, it would likely require a new version of those endpoints).

Reply:
Ah indeed, too risky, no worries.